### ISSUE TITLE
Nativescript setInterval does not run when app is not in focus

### ISSUE THREAD

**Issue by dpdragnev** [NativeScript/nativescript-angular#868]
Hello,

I  have a NS 3.0 + angular + TS application. In it, I am setting an interval and I am getting some data from the server when the interval elapses.

Everything runs fine in the simulator. If I go to the home screen (with the app running on the background) the data is being pulled correctly. However, on the device the interval function fires only if the app is active. I am testing on iPhone 5s (iOS 10.3.2).

How can I resolve this?

P.S. Every interval, If there is new data I create a local notification using https://github.com/EddyVerbruggen/nativescript-local-notifications. On the simulator, I can even lock the phone and the notifications still appear. On the device this works only if the app is active. Not sure if this makes any difference, but I just wanted to mention it.

Thanks.

**Comment by proweber1** [comment]
Have any solutions? I have same problem

**Comment by dpdragnev** [comment]
Not really.  I have not found a way to run it on iOS.  At least not using Nativescript, maybe there is a way to write some native code to accomplish that, I am not sure.  

I ended up running the interval on the server and just pushed the notification to the device when needed.  This actually worked pretty well, so I am sticking with this approach.
